services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -498,27 +505,37 @@ async def _get_project_lock_state(
         return ProjectLocked(value=is_locked)
 
 
-async def _get_project_running_state(
-    user_id: PositiveInt, project_uuid: str, app: web.Application
-) -> ProjectRunningState:
-    pipeline_state: RunningState = await get_pipeline_state(app, user_id, project_uuid)
-    return ProjectRunningState(value=pipeline_state)
+async def add_project_states_for_user(
+    user_id: int, project: Dict[str, Any], is_template: bool, app: web.Application
+) -> Dict[str, Any]:
 
+    lock_state = ProjectLocked(value=False)
+    running_state = RunningState.UNKNOWN
+    if not is_template:
+        lock_state, computation_task = await logged_gather(
```

Comment:
if requests to directorv2 fail, this will raise exception and you cannot get the current state 
How should it be handled by the callers? I guess in some cases, the caller might "live with it" and set some kind of "unknown" state but others cannot resolve without this info and reraise another error.

It is important to set an error policy in all functions in ``projects_api.py`` since it is the entry point to couple with the projects module's app (kind of the API of the future project's service)